src/PIL/Image.py

```diff
@@ -2137,11 +2137,11 @@ def thumbnail(self, size, resample=BICUBIC):
         # preserve aspect ratio
         x, y = self.size
         if x > size[0]:
-            y = int(max(y * size[0] / x, 1))
-            x = int(size[0])
+            y = max(round(y * size[0] / x), 1)
+            x = size[0]
```

Comment:
Just to be aware - by removing the `int` rounding on these second lines, if someone had existing code `im.thumbnail((150.2, 150.2))`, it would now break.

That's not unreasonable though.

Reply:
Example:

```python
from PIL import Image

im = Image.open("Tests/images/hopper.png")
print(im.size)

im.thumbnail((15.2, 15.2))
print(im.size)

im.thumbnail((10, 10))
print(im.size)
```

On 6.2.1:

```
(128, 128)
(15, 15)
(10, 10)
```

On master with this PR:

```
(128, 128)
Traceback (most recent call last):
  File "1.py", line 6, in <module>
    im.thumbnail((15.2, 15.2))
  File "/Users/hugo/github/Pillow/src/PIL/Image.py", line 2153, in thumbnail
    im = self.resize(size, resample)
  File "/Users/hugo/github/Pillow/src/PIL/Image.py", line 1822, in resize
    return self._new(self.im.resize(size, resample, box))
TypeError: integer argument expected, got float
```

Should we fix it something like this?

```diff
diff --git a/src/PIL/Image.py b/src/PIL/Image.py
index 613b3a36..981c8c81 100644
--- a/src/PIL/Image.py
+++ b/src/PIL/Image.py
@@ -2138,10 +2138,10 @@ class Image:
         x, y = self.size
         if x > size[0]:
             y = max(round(y * size[0] / x), 1)
-            x = size[0]
+            x = round(size[0])
         if y > size[1]:
             x = max(round(x * size[1] / y), 1)
-            y = size[1]
+            y = round(size[1])
         size = x, y

         if size == self.size:
```

```
(128, 128)
(15, 15)
(10, 10)
```

Reply:
Okay, I've created PR #4272